packages/react-native-sessions/src/contexts/SessionContext.tsx

```diff
@@ -0,0 +1,390 @@
+        const [walletsResponse, userResponse] = await Promise.all([
+          client.getWallets({ organizationId: whoami.organizationId }),
+          client.getUser({
+            organizationId: whoami.organizationId,
+            userId: whoami.userId,
+          }),
+        ]);
+
+        const wallets = await Promise.all(
+          walletsResponse.wallets.map(async (wallet) => {
+            const accounts = await client.getWalletAccounts({
+              organizationId: whoami.organizationId,
+              walletId: wallet.walletId,
+            });
+            return {
+              name: wallet.walletName,
+              id: wallet.walletId,
+              accounts: accounts.accounts.map((account) => {
+                return {
+                  id: account.walletAccountId,
+                  curve: account.curve,
+                  pathFormat: account.pathFormat,
+                  path: account.path,
+                  addressFormat: account.addressFormat,
+                  address: account.address,
+                  createdAt: account.createdAt,
+                  updatedAt: account.updatedAt,
+                };
+              }),
+            };
+          }),
+        );
+
+        const user = userResponse.user;
+
+        setUser({
+          id: user.userId,
+          userName: user.userName,
+          email: user.userEmail,
+          phoneNumber: user.userPhoneNumber,
+          organizationId: whoami.organizationId,
+          wallets,
+        });
+      }
+    }
+  };
+
+  /**
+   * Refreshes the user state.
+   *
+   * - Calls `fetchUser` to update user data.
+   * - Should be run when user data changes.
+   */
+  const refreshUser = async () => {
+    await fetchUser();
+  };
+
+  /**
+   * Clears any scheduled expiration timeouts
+   */
+  const clearTimeouts = () => {
+    if (expiryTimeoutRef.current) clearTimeout(expiryTimeoutRef.current);
+  };
+
+  /**
+   * Schedules session expiration callback
+   * @param expiryTime - The Unix timestamp of session expiration
+   */
+  const scheduleSessionExpiration = (expiryTime: number) => {
+    clearTimeouts();
+    const timeUntilExpiry = expiryTime - Date.now();
+
+    if (timeUntilExpiry > 0) {
+      expiryTimeoutRef.current = setTimeout(() => {
+        clearSession();
+        config.onSessionExpired?.();
+      }, timeUntilExpiry);
+    } else {
+      clearSession();
+      config.onSessionExpired?.();
+    }
+  };
+
+  /**
+   * Retrieves the stored embedded key from secure storage.
+   * Optionally deletes the key from storage after retrieval.
+   *
+   * @param deleteKey Whether to remove the embedded key after retrieval. Defaults to `true`.
+   * @returns The embedded private key if found, otherwise `null`.
+   * @throws If retrieving or deleting the key fails.
+   */
+  const getEmbeddedKey = async (deleteKey = true) => {
+    const credentials = await Keychain.getGenericPassword({
+      service: "turnkey-embedded-key",
```

Comment:
Could we turn this into constant defined in the constants.ts file? and then replace instances of it in this file with the constant?